client/src/app/shared/menu/top-menu-dropdown.component.ts

```diff
@@ -23,38 +32,49 @@ export type TopMenuDropdownParam = {
   templateUrl: './top-menu-dropdown.component.html',
   styleUrls: [ './top-menu-dropdown.component.scss' ]
 })
-export class TopMenuDropdownComponent implements OnInit, OnDestroy {
+export class TopMenuDropdownComponent implements OnInit, OnDestroy, AfterViewInit {
   @Input() menuEntries: TopMenuDropdownParam[] = []
 
+  @ViewChild('modal', { static: true }) modal: ElementRef
```

Comment:
```suggestion
  @ViewChild('modal', { static: true }) modal: NgbModal
```

Reply:
I know we are not very consistent with this throughout the application, but that gives a bit more info type-wise.